app/scripts/app-init.js

```diff
@@ -189,6 +189,24 @@ const registerInPageContentScript = async () => {
   }
 };
 
+// this polyfill can be removed once we drop support for Firefox v121 (after
```

Comment:
I would just use our existing `deferredPromise` helper, but `app-init.js` doesn't `import` anything currently (and i'm not sure if it even can?)